--- /Component/MassAction/Filter.php
+++ /Component/MassAction/Filter.php
@@ -80,16 +80,6 @@ public function getComponent()
      */
     public function getCollection(AbstractDb $collection)
     {
-        $component = $this->getComponent();
-        $this->prepareComponent($component);
-        $dataProvider = $component->getContext()->getDataProvider();
-        $dataProvider->setLimit(0, false);
-        $ids = [];
-        foreach ($dataProvider->getSearchResult()->getItems() as $document) {
-            $ids[] = $document->getId();
-        }
-
-        $collection->addFieldToFilter($collection->getIdFieldName(), ['in' => $ids]);
         return $this->applySelection($collection);
     }